### ISSUE TITLE
ember/no-shadow-route-definition false positive

### ISSUE THREAD

**Issue by BlueRaja** [ember-cli/eslint-plugin-ember#1037]
The following is triggering the rule incorrectly:
```
  this.route('edit', { path: ':experienceId' }, function () {
    this.route('views', { path: 'views/:viewId' });
    this.route('viewcollections', { path: 'viewcollections/:viewCollectionId' });
  });
```

**Comment by raido** [comment]
I'll try to take a look at this. I have other questionable cases in notes as well, example conditional route definitions.

```
if (X) {
   this.route(...)
}
```

**Comment by raido** [comment]
I opened PR for specifically false positive case mentioned in this issue.
